avalon/tools/widgets.py

```diff
@@ -61,64 +47,50 @@ def __init__(self, silo_creatable=True, parent=None):
         view.setHeaderHidden(True)
         view.setModel(proxy)
 
+        # Header
+        header = QtWidgets.QHBoxLayout()
+
+        icon = qtawesome.icon("fa.refresh", color=style.colors.light)
+        refresh = QtWidgets.QPushButton(icon, "")
+        refresh.setToolTip("Refresh items")
+
         filter = QtWidgets.QLineEdit()
         filter.textChanged.connect(proxy.setFilterFixedString)
         filter.setPlaceholderText("Filter assets..")
 
+        header.addWidget(filter)
+        header.addWidget(refresh)
+
         # Layout
         layout.addLayout(header)
-        layout.addWidget(filter)
         layout.addWidget(view)
 
         # Signals/Slots
         selection = view.selectionModel()
         selection.selectionChanged.connect(self.selection_changed)
         selection.currentChanged.connect(self.current_changed)
-        silo.silo_changed.connect(self._on_silo_changed)
         refresh.clicked.connect(self.refresh)
 
         self.refreshButton = refresh
-        self.silo = silo
         self.model = model
         self.proxy = proxy
         self.view = view
 
-    def _on_silo_changed(self):
-        """Callback for silo change"""
-
-        self._refresh_model()
-        silo = self.get_current_silo()
-        self.silo_changed.emit(silo)
-        self.selection_changed.emit()
-
     def _refresh_model(self):
-
-        silo = self.get_current_silo()
-        with lib.preserve_expanded_rows(self.view,
-                                        column=0,
-                                        role=self.model.ObjectIdRole):
-            with lib.preserve_selection(self.view,
-                                        column=0,
-                                        role=self.model.ObjectIdRole):
-                self.model.set_silo(silo)
+        with lib.preserve_states(
+            self.view, column=0, role=self.model.ObjectIdRole
+        ):
+            self.model.refresh()
 
         self.assets_refreshed.emit()
 
     def refresh(self):
-
-        silos = _list_project_silos()
-        self.silo.set_silos(silos)
-
         self._refresh_model()
 
-    def get_current_silo(self):
-        """Returns the currently active silo."""
-        return self.silo.get_current_silo()
-
     def get_active_asset(self):
         """Return the asset id the current asset."""
```

Comment:
Is this still returning the `id` of the document with the changed role? It seems like this docstring might be wrong currently.